blocks/model.py

```diff
@@ -161,15 +162,18 @@ def __init__(self, outputs):
             raise ValueError("top bricks with the same name:"
                              " {}".format(', '.join(repeated_names)))
 
-        brick_param_names = {
-            v: k for k, v in Selector(self.top_bricks).get_params().items()}
-        self.params = []
-        for param in VariableFilter(roles=[PARAMETER])(self.shared_variables):
-            if param in brick_param_names:
-                self.params.append((brick_param_names[param], param))
+        brick_parameter_names = {
+            v: k
+            for k, v in Selector(self.top_bricks).get_parameters().items()}
+        self.parameters = []
```

Comment:
Oh! There is a conflict with ComputationGraph.parameters (see [here](https://github.com/mila-udem/blocks/blob/master/blocks/graph.py#L92)). It seems Model previously had both params and parameters attributes. 

You can use get, set interface to get around this. Build parameters locally (without self.) and then use set_paramter_values(). 

However there is still a problem. There is a get, set interface for parameters **plus** an attribute parameters which is built differently and returns a list rather than an OrderedDict. I think ComputationGraph.parameters should be overridden. @bartvm @rizar @dwf Which one would you keep?

To override ComputationGraph.parameters we would have to define getter method with @property and hide parameters under _parameters. That means we will have 2 getter functions for _parameters. I don't like that much, why not using getter and setter from @property all over for attribute parameters? Or there is a better way to override it that I don't know?

![](http://cdn.meme.am/instances/500x/56094413.jpg)